chia/full_node/coin_store.py

```diff
@@ -53,10 +53,16 @@ async def create(cls, db_wrapper: DBWrapper, cache_size: uint32 = uint32(60000))
             "CREATE INDEX IF NOT EXISTS coin_confirmed_index on coin_record(confirmed_index)"
         )
 
-        await self.coin_record_db.execute("CREATE INDEX IF NOT EXISTS coin_spent_index on coin_record(spent_index)")
-
         if self.db_wrapper.allow_upgrades:
             await self.coin_record_db.execute("DROP INDEX IF EXISTS coin_spent")
+            await self.coin_record_db.execute("DROP INDEX IF EXISTS coin_spent_index")
+            await self.coin_record_db.execute(
+                "CREATE INDEX IF NOT EXISTS coin_spent_idx on coin_record(spent_index) WHERE spent_index>0"
+            )
+        else:
+            await self.coin_record_db.execute(
+                "CREATE INDEX IF NOT EXISTS coin_spent_index on coin_record(spent_index) WHERE spent_index>0"
```

Comment:
note that this else-case is run both by people not opting into `allow_upgrades` but also by new users. We need to use the existing name of this index, to not cause a re-index for existing users, but we still use the new condition, to make new users use the smaller index.